src/connection/impls/file.rs

```diff
@@ -105,6 +106,58 @@ mod ioctl {
 
     ioctl_readwrite!(ipmi_recv_msg_trunc, IPMI_IOC_MAGIC, 11, IpmiRecv);
     ioctl_read!(ipmi_send_request, IPMI_IOC_MAGIC, 13, IpmiRequest);
+    ioctl_read!(ipmi_get_my_address, IPMI_IOC_MAGIC, 18, u32);
+}
+
+#[repr(C)]
+enum IpmiAddr {
+    SysIface(IpmiSysIfaceAddr),
+    Ipmb(IpmiIpmbAddr),
+}
+
+impl IpmiAddr {
+    fn ptr(&mut self) -> *mut u8 {
+        match self {
+            IpmiAddr::SysIface(ref mut bmc_addr) => std::ptr::addr_of_mut!(*bmc_addr) as *mut u8,
+            IpmiAddr::Ipmb(ref mut ipmb_addr) => std::ptr::addr_of_mut!(*ipmb_addr) as *mut u8,
+        }
+    }
+    fn size(&self) -> u32 {
+        match self {
+            IpmiAddr::SysIface(_) => core::mem::size_of::<IpmiSysIfaceAddr>() as u32,
+            IpmiAddr::Ipmb(_) => core::mem::size_of::<IpmiIpmbAddr>() as u32,
+        }
+    }
+}
+
+impl From<RequestTargetAddress> for IpmiAddr {
+    fn from(value: RequestTargetAddress) -> Self {
+        match value {
+            RequestTargetAddress::Bmc(lun) => {
+                IpmiAddr::SysIface(IpmiSysIfaceAddr::bmc(lun.value()))
+            }
+            RequestTargetAddress::BmcOrIpmb(addr, channel, lun) => {
+                IpmiAddr::Ipmb(IpmiIpmbAddr::new(channel.0 as i16, addr.0, lun.value()))
+            }
+        }
+    }
+}
+
+impl Display for IpmiAddr {
```

Comment:
:+1: I always forget that implementing `Display` is so trivial, this is a lot nicer